fem/bilininteg.hpp

```diff
@@ -2129,6 +2157,23 @@ class DiffusionIntegrator: public BilinearFormIntegrator
                                        ElementTransformation &Trans,
                                        DenseMatrix &elmat);
 
+   virtual void AssemblePatchMatrix(const int patch, Mesh *mesh,
+                                    SparseMatrix*& smat);
+
+   void AssemblePatchMatrix_fullQuadrature(const int patch, Mesh *mesh,
+                                           SparseMatrix*& smat);
+
+   void AssemblePatchMatrix_reducedQuadrature(const int patch, Mesh *mesh,
+                                              SparseMatrix*& smat);
+
+   void AssemblePatchMatrix_simpleButInefficient(const int patch,
+                                                 Mesh *mesh,
+                                                 DenseMatrix &pmat);
```

Comment:
Brief comments would be useful. Particularly since "patch" might be used on non-NURBS contexts and therefore be confusing to some readers.